mautic/mautic
# Create user with automatic credentials and email for 5.x

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | [mautic/mautic-documentation#... ](https://github.com/mautic/mautic-documentation/pull/181)
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | N/A



This is the same pr of https://github.com/mautic/mautic/pull/9733 but for Mautic 5.0 alpha release.


Added new method to create a new user.

I added a checkbox  "Generate random password and send an email with credentials to user. ".

If it is checked, inputs password will be hidden and the password will be automatically generated, 
in addition, an email with the credentials will be sent to the user.

![newusermautic](https://user-images.githubusercontent.com/50916237/109299020-ba0d5300-7834-11eb-9ee6-e02a26536a30.png)




#### Steps to test this PR:
1. Checkout to the branch of this PR
2. Generate new assets with `php bin/console mautic:assets:generate` command and clear cache with `php bin/console cache:clear` command
3. Configure email settings
4. Go to new user page
5. Check "Generate random password and send an email with credentials to user. "
6. Wait for email
